src/Microsoft.Health.Fhir.Api/Features/Filters/OperationOutcomeExceptionFilterAttribute.cs

```diff
@@ -87,20 +94,53 @@ public override void OnActionExecuted(ActionExecutedContext context)
                     case UnsupportedConfigurationException _:
                         fhirResult.StatusCode = HttpStatusCode.InternalServerError;
                         break;
+                }
+
+                context.Result = fhirResult;
+                context.ExceptionHandled = true;
+            }
+            else if (context.Exception is MicrosoftHealthException microsoftHealthException)
+            {
+                FhirResult healthExceptionResult;
+
+                switch (microsoftHealthException)
+                {
                     case RequestRateExceededException ex:
-                        fhirResult.StatusCode = HttpStatusCode.TooManyRequests;
+                        healthExceptionResult = FhirResult.Create(
+                            new OperationOutcome
+                            {
+                                Id = _fhirRequestContextAccessor.FhirRequestContext.CorrelationId,
+                                Issue = new List<OperationOutcome.IssueComponent>
+                                {
+                                    new OperationOutcome.IssueComponent
+                                    {
+                                        Severity = OperationOutcome.IssueSeverity.Error,
+                                        Code = OperationOutcome.IssueType.Throttled,
+                                        Diagnostics = ex.Message,
+                                    },
+                                },
+                            }, HttpStatusCode.BadRequest);
+                        healthExceptionResult.StatusCode = HttpStatusCode.TooManyRequests;
 
                         if (ex.RetryAfter != null)
                         {
-                            fhirResult.Headers.Add(
+                            healthExceptionResult.Headers.Add(
                                 RetryAfterHeaderName,
                                 ex.RetryAfter.Value.TotalMilliseconds.ToString(CultureInfo.InvariantCulture));
                         }
 
+                        break;
+                    default:
+                        healthExceptionResult = FhirResult.Create(
+                            new OperationOutcome
+                            {
+                                Id = _fhirRequestContextAccessor.FhirRequestContext.CorrelationId,
+                            }, HttpStatusCode.BadRequest);
+                        healthExceptionResult.StatusCode = HttpStatusCode.BadRequest;
```

Comment:
should this be internal server error since it's an exception that we couldn't "handle"